gh-115315: Update time.rst to include microseconds field (%f) in chart

TahomaSoft opened the pull request:
Added documentation on microseconds and the %f field.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNN: Summary of the changes made
```

Where: gh-NNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-115315 -->
* Issue: gh-115315
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--115316.org.readthedocs.build/en/115316/library/time.html

<!-- readthedocs-preview cpython-previews end -->

TahomaSoft committed:
Update time.rst

Added documentation on microseconds and the %f field.

hugovk commented on Doc/library/time.rst:
```diff
@@ -454,6 +454,10 @@ Functions
    | ``%S``    | Second as a decimal number [00,61].            | \(2)  |
    |           |                                                |       |
    +-----------+------------------------------------------------+-------+
+   | ``%f`     | Microseconds as a decimal number               |       | 
```
Build and lint fixes:

```suggestion
   | ``%f``    | Microseconds as a decimal number               |       |
```

hugovk reviewed (commented):
See also https://github.com/python/cpython/issues/115315#issuecomment-1938596682.

TahomaSoft committed:
Update Doc/library/time.rst

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

TahomaSoft committed:
Merge branch 'python:main' into time-doc

TahomaSoft committed:
Update time.rst

Made changes on table with note regarding strptime and strftime per conversation with Hugo.

hugovk commented on Doc/library/time.rst:
```diff
@@ -510,7 +514,11 @@ Functions
-   (3)
+   (3) 
+       The %f format directive only applies to time.strptime, not to time.strftime. 
+       However, see also datetime.strptime and datetime.strftime.
```
Remove trailing whitespace, link to functions, wrap to 80 cols:

```suggestion
       The ``%f` format directive only applies to :func:`strptime`,
       not to :func:`strftime`.
       However, see also :meth:`datetime.strptime` and
       :meth:`datetime.strftime`.
```

TahomaSoft committed:
Update Doc/library/time.rst

Thanks for the tweaks!

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

hugovk commented on Doc/library/time.rst:
```diff
@@ -510,7 +514,13 @@ Functions
+       The ``%f` format directive only applies to :func:`strptime`,
+       not to :func:`strftime`.
+       However, see also :meth:`datetime.strptime` and
+       :meth:`datetime.strftime`.
```
Sorry, I missed a backtick, and we need the full ref for datetime:
```suggestion
   (3)
       The ``%f`` format directive only applies to :func:`strptime`,
       not to :func:`strftime`.
       However, see also :meth:`datetime.datetime.strptime` and
       :meth:`datetime.datetime.strftime`.
```

TahomaSoft committed:
Update Doc/library/time.rst

Thanks you

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

TahomaSoft committed:
Merge branch 'main' into time-doc

TahomaSoft committed:
Merge branch 'main' into time-doc

hugovk commented on Doc/library/time.rst:
```diff
@@ -511,6 +515,12 @@ Functions
+       The ``%f`` format directive only applies to :func:`strptime`,
+       not to :func:`strftime`.
+       However, see also :meth:`datetime.datetime.strptime` and
+       :meth:`datetime.datetime.strftime`.
```
This doesn't really tell us much. Why should we see those datetime methods?

By the way, please don't update the PR from `main` unless there's something important we need from there, or a conflict to resolve. It needlessly uses up extra CI time. Thanks!

TahomaSoft replied:
Ok, thanks for letting me know about the SOP (standard operating procedure) here about not updating the PR. Sorry about that. Other entities (some embedded Linux people) do want it updated.

How about:
However, see also :meth:`datetime.datetime.strptime` and
       :meth:`datetime.datetime.strftime` where the %f format directive applies to both methods in order to handle microseconds.

Alternatively, the whole "However ... " clause could be deleted.

hugovk replied:
> Ok, thanks for letting me know about the SOP (standard operating procedure) here about not updating the PR. Sorry about that. Other entities (some embedded Linux people) do want it updated.

No problem! Yeah, projects vary a lot :)

> How about: However, see also :meth:`datetime.datetime.strptime` and :meth:`datetime.datetime.strftime` where the %f format directive applies to both methods in order to handle microseconds.

Sounds good, how about:

```rst
However, see also :meth:`datetime.datetime.strptime` and
:meth:`datetime.datetime.strftime` where the ``%f`` format directive
:ref:`applies to microseconds <format-codes>`.
```

TahomaSoft replied:
Thanks!  Works for me. I'll make that change to my branch now.

TahomaSoft replied:
Sorry, didn't realize when I hit "resolve conversation" the CI checks would run again....

hugovk replied:
"Resolve" doesn't trigger the CI, pushing does -- and we want it to trigger for new pushes. Thanks!

TahomaSoft committed:
Update time.rst

Made changes requested to put the datetime references in better context.

hugovk reviewed (approved):
Thanks!

Let's give some time for the CODEOWNERS to review before merging.

willingc reviewed (commented):
Thanks @TahomaSoft. While we wait on feedback from the CODEOWNERS, it would be good to follow the alphabetical pattern of the table and move `%f` up in the table.

TahomaSoft committed:
alphabetically reordered format labels per request. Adjusted notes numbers and sequentce to match
